main/plots.py

```diff
@@ -387,10 +247,84 @@ def create_capacity_planning_layout() -> Row:
         end_picker=end_picker,
     )
 
-    # Create layout to display widgets aligned as a column next to the plot
-    plot_layout = row(
-        column(start_picker, end_picker, calendar_button, financial_button), plot
+    # Project(s) for selection
+    project_options = all_projects
+    project_title = Div(text="<h3>Select Projects</h3>", width=180)
+    project_multichoice = MultiChoice(
+        options=[(opt, opt) for opt in project_options],
+        value=[],  # empty default implies all projects efforts selected
+        width=180,  # width of the multichoice fr projects
+    )
+
+    # User(s) selection
+    user_options = all_users
+    user_title = Div(text="<h3>Select Users</h3>", width=180)
+    user_multichoice = MultiChoice(
+        options=[(opt, opt) for opt in user_options],
+        value=[],  # empty default implies all users capacity selected
+        width=180,  # width of the multichoice for users
     )
+
+    # Button to clear all filters and reset plot to default state
+    reset_project_button = Button(label="Reset Projects", width=100)
+    reset_user_button = Button(label="Reset Users", width=100)
+
+    # combined callback
+    callback_code = CustomJS(
+        args=dict(
+            plot=plot,
+            project_multichoice=project_multichoice,
+            user_multichoice=user_multichoice,
+            start_picker=start_picker,
+            end_picker=end_picker,
+            min_date=min_date,
+            max_date=max_date,
+        ),
+        code="""
+            console.log("Selected projects:", project_multichoice.value);
+            console.log("Selected users:", user_multichoice.value);
+            plot.change.emit();
```

Comment:
This does not result in the plot being re-drawn with the new data as, from the client side point of view, there's no different data. Indeed, this is going to be challenging since for this to work, it is necessary to call the python code that create the traces and get that new data into the client. 

I don't have an answer about that. You could check [this discussion](https://stackoverflow.com/questions/71191122/bokeh-generate-graph-server-side-update-graph-from-js-client-side-change-data) for inspiration - maybe? - and have a chat with @stephwills , who might have found something similar previously. If there's no obvious solution, maybe we can have a chat later today or tomorrow.

Reply:
Thanks! I will take a look at that link and happy to chat later too.

Reply:
@stephwills thanks! Although this workaround requires pre-loading all the project and user data.